sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -236,11 +214,13 @@ namespace Azure { namespace Core { namespace Test {
       return std::make_unique<T>(url, credential, options);
     }
 
+    template <class T> void InitClientOptions(T& options) { PrepareOptions(options); }
```

Comment:
what's the point ? toghether with the next change ?

Reply:
This is useful when we already have an options, and we just want to append the record-playback policy into the pipeline.

Reply:
PrepareOptions is getting wrapped in InitClientOptions method that only calls the PrepareOptions. i don't see the benefit of this construct , what does the indirection provide vs calling PrepareOptions alone?

Reply:
`PrepareOptions` is a private method, thus cannot be called in inherited classes.

Reply:
then you could move it to protected

Reply:
`PrepareOptions` is used in multiple places. It's mainly for internal implementation. I don't feel comfortable exposing it to derived classes.